### PR TITLE
fix aperçu/dl fichier quand il est dans {userid}/temp

### PR DESCRIPTION
**Pull request by atm-saamiperdrix**

### CODE REVIEW

**Comment by eldy** [comment]
Can you describe process how to reproduce the bug ?

**Comment by atm-saamiperdrix** [comment]
Ajouter une PJ manuellement puis essayer de la voir / télécharger

File: htdocs/document.php [hunk 1]
```diff
@@ -207,19 +207,18 @@ function llxFooter()
...
+//$refname = basename(dirname($original_file)."/");
 
 // Security check
 if (empty($modulepart)) {
 	accessforbidden('Bad value for parameter modulepart');
 }
 
 // Check security and set return info with full path of file
-$check_access = dol_check_secure_access_document($modulepart, $original_file, $entity, $user, $refname);
+$check_access = dol_check_secure_access_document($modulepart, $original_file, $entity, $user, '');
```
Comment (eldy) [inline htdocs/document.php:218]: This introduce a security hole by allowing to download file without checking on object itself (just on permission on module) so no check on entity become possible.

**Comment by eldy** [comment]
On a particular object ? Can you reproduce the bug on v19 branch ?

**Comment by atm-quentin** [comment]
For example, on a commercial proposal, when we attach a file and want to view it through the magnifying glass, if we don't have the "View other users, groups and their permissions" right, we won't be able to view it even though we've just attached it ourselves.

**Comment by atm-quentin** [comment]
i reproduced on v19

**Comment by atm-quentin** [comment]
in an email form

**Pull request merged**